firepit/welcome.js

```diff
@@ -0,0 +1,213 @@
+    .map(() => "-")
+    .join("");
+  console.log(line);
+  console.log("");
+  console.log("     Welcome to...");
+  console.log(chalk.yellow(banner));
+  console.log("");
+  console.log(line);
+  console.log("");
+  await CheckFirebaseTools();
+  console.log("");
+  await CheckIsLoggedIn();
+  console.log("");
+  console.log(
+    `${chalk.green("+")} You can now use the 'firebase' or 'npm' commands!`
+  );
+  console.log(
+    `${chalk.blue("~")} For more help see https://firebase.google.com/docs/cli/`
+  );
+  console.log("");
+  console.log(thin_line);
+  process.exit();
+})();
+
+async function CheckFirebaseTools() {
+  console.log(
+    `${chalk.cyan("~")} Let's make sure your Firebase CLI is ready...`
+  );
+  firebase_bin = await GetFirebaseToolsBins();
+  const isSetup = !!firebase_bin;
+
+  if (isSetup) {
+    console.log(
+      `${chalk.green("+")} Looks like ${chalk.green("your CLI is set up")}!`
+    );
+  } else {
+    console.log(
+      `${chalk.cyan("~")} Looks like ${chalk.yellow(
+        "your CLI needs to be set up"
+      )}.\n`
+    );
+    console.log(`${chalk.cyan("~")} This may take a moment`);
+    process.stdout.write(`${chalk.green("+")} Setting up environment...`);
+
+    return new Promise(resolve => {
+      const exe_split = firebase_exe.split(" ");
+      const install = spawn(
+        exe_split[0],
+        [...exe_split.slice(1), "--tool:force-setup"],
+        {}
+      );
+
+      install.stderr.on("data", buf => {
+        readline.clearLine(process.stdout, 2);
+        readline.cursorTo(process.stdout, 2, null);
+        const line = (
+          buf
+            .toString()
+            .split("\n")
+            .filter(l => l)
+            .slice(-1)[0] || ""
+        )
+          .trim()
+          .slice(0, process.stdout.columns - 5);
+        process.stdout.write(line);
+      });
+
+      install.on("exit", () => {
+        readline.clearLine(process.stdout, 0);
+        readline.cursorTo(process.stdout, 0, null);
+        process.stdout.write(
+          `${chalk.green("+")} Alright, ${chalk.green("your CLI is set up")}!\n`
+        );
+        resolve();
+      });
+    });
+  }
+}
+
+async function GetFirebaseToolsBins() {
+  return new Promise(resolve => {
+    const exe_split = firebase_exe.split(" ");
+    const checkSpawn = spawn(exe_split[0], [
+      ...exe_split.slice(1),
+      "--tool:setup-check"
+    ]);
+    let checkSpawnData = "";
+
+    checkSpawn.stdout.on("data", buf => {
+      checkSpawnData += buf;
+    });
+
+    checkSpawn.on("close", () => {
+      firebase_bin = JSON.parse(checkSpawnData).bins[0];
+      resolve(firebase_bin);
+    });
+  });
+}
+
+async function CheckIsLoggedIn() {
+  firebase_bin = firebase_bin || (await GetFirebaseToolsBins());
+  return new Promise(resolve => {
+    const finish = SimpleSpinner(
+      `${chalk.cyan("~")} Checking your Firebase credentials`
+    );
+    const listSpawn = fork(firebase_bin, ["list", "--json"], { silent: true });
```

Comment:
FYI I think they're getting rid of the `list` command soon.

Reply:
Oh, hmm, well this can really be any no-op that requires auth. It just needs to fail out if we're not authed so it's probably fine to leave until this point.